app/frontend/src/features/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { Context, ReactChild, useContext } from "react";
+import useAuthStore, { AuthStoreType } from "./useAuthStore";
+
+export const AuthContext = React.createContext<null | AuthStoreType>(null);
+
+export function useAppContext<T>(context: Context<T | null>) {
+  const contextValue = useContext(context);
+  if (contextValue === null) {
+    throw Error("No context provided!");
+  }
+  return contextValue;
+}
```

Comment:
```suggestion
const AuthContext = React.createContext<null | AuthStoreType>(null);

export function useAppContext() {
  const contextValue = useContext(AuthContext);
  if (contextValue === null) {
    throw Error("No context provided!");
  }
  return contextValue;
}
```
Presumably `AppContext` is only gonna be used with this helper hook, you could do this instead. Then everywhere else in the app, you can `useAppContext` without needing `AuthContext` from this file and so it doesn't even need to be exported.